src/celeritas/global/detail/AlongStepActionImpl.hh

```diff
@@ -186,8 +199,16 @@ inline CELER_FUNCTION void along_step_track(CoreTrackView const& track)
     }
 
     using Energy = ParticleTrackView::Energy;
-    Energy eloss = calc_energy_loss(
-        track.make_cutoff_view(), mat, particle, phys, step_limit.step, rng);
+    Energy eloss
+        = (particle.energy() < phys.scalars().eloss_calc_limit && !geo_limited)
+              ? particle.energy()
+              : calc_energy_loss(track.make_cutoff_view(),
+                                 mat,
+                                 particle,
+                                 phys,
+                                 step_limit.step,
+                                 rng);
+
```

Comment:
I think we also need to check that this is a particle with an energy loss process so we don't apply this cut to photons.

Reply:
You are absolutely right! Probably we want to move this threshold check back into the calc_energy_loss (without the geo_limited condition?).  Actually, this lowest energy cutoff reduced around 1/3 of number of steps before fixing the true_path, but did not make much difference after the fix (i.e, order dependent) as you did not see much impact of the approximation.  So, another choice may be to move back to the original implementation without any conditional check.

Reply:
We do want to keep the geo-limited condition to prevent tracks from stopping on a boundary, right?

Reply:
Not sure whether we still need this extra safe guard - I successfully tested demo-loop with testem3-flat and 100k input file without any problem (and of course, simple-cms  and the hepmc3 input).  So, let's drop it for now until any exceptional case is happened or do you have a new failure because of dropping the condition?

Reply:
ok, let's leave it as Is :)

Reply:
(This would only show up as an assertion failure with orange and if the stopped particle was a positron, when the secondary photons try to initialize on a surface.)

For reference, looking at testem3 with 100k primaries, msc enabled, and with this cut, there are ~500 tracks coming to rest on a boundary (with or without energy loss fluctuations, but slightly more with). Without fluctuations only a handful of those are positrons; with fluctuations ~20% are.

Reply:
Yuck, that's not good... is there an expedient solution? I'm also in the testing phase of the along-step refactor (about to open up a draft pull request for comment now) so we may be able to more cleanly add this cutoff with the new structure?

Reply:
Not sure what to do with the stopped particles on a boundary from eloss fluctuations, so yeah let's leave it as is here and work on it in your along-step refactor.